demos/burgers-isotropic.py

```diff
@@ -0,0 +1,274 @@
+    end_time,
+    num_subintervals,
+    dt,
+    fields,
+    num_timesteps_per_export=2,
+)
+
+params = GoalOrientedMetricParameters(
+    {
+        "element_rtol": 0.001,
+        "maxiter": 35 if os.environ.get("GOALIE_REGRESSION_TEST") is None else 3,
+    }
+)
+
+mesh_seq = GoalOrientedMeshSeq(
+    time_partition,
+    meshes,
+    get_function_spaces=get_function_spaces,
+    get_initial_condition=get_initial_condition,
+    get_form=get_form,
+    get_solver=get_solver,
+    get_qoi=get_qoi,
+    qoi_type="time_integrated",
+    parameters=params,
+)
+
+# As in the previous adaptation demos, the most important part is the adaptor function.
+# The one used here takes a similar form, except that we need to handle multiple meshes
+# and metrics.
+# We then time integrate these to give the metric contribution from each subinterval.
+# Given that we use a simple implicit Euler method for time integration
+# in the PDE, we do the same here, too.
+#
+# Goalie provides functionality to normalise a list of metrics using *space-time*
+# normalisation. This ensures that the target complexity is attained on average across
+# all timesteps.
+#
+# Note that when adapting the mesh, we need to be careful to check whether convergence
+# has already been reached on any of the subintervals. If so, the adaptation step is
+# skipped. ::
+
+
+def adaptor(mesh_seq, solutions, indicators):
+    metrics = []
+    complexities = []
+
+    # Ramp the target average metric complexity per timestep
+    base, target, iteration = 400, 1000, mesh_seq.fp_iteration
+    mp = {
+        "dm_plex_metric": {
+            "target_complexity": ramp_complexity(base, target, iteration),
+            "p": 1.0,
+            "h_min": 1.0e-04,
+            "h_max": 1.0,
+        }
+    }
+
+    # Construct the metric on each subinterval
+    for i, mesh in enumerate(mesh_seq):
+        sols = solutions["u"]["forward"][i]
+        dt = mesh_seq.time_partition.timesteps[i]
+
+        # Define the Riemannian metric
+        P1_ten = TensorFunctionSpace(mesh, "CG", 1)
+
+        # At each timestep, recover metric of the solution
+        # vector. Then time integrate over the contributions
+        _metrics = []
+
+        for j, sol in enumerate(sols):
+
+            # get local indicator
+            indi = indicators["u"][i][j]
+            # local instance of Riemanian metric
+            _metric = RiemannianMetric(P1_ten)
+
+            # reset parameters
+            _metric.set_parameters(mp)
+
+            # Deduce an isotropic metric from the error indicator field
+            _metric.compute_isotropic_metric(error_indicator=indi, interpolant="L2")
+            _metric.normalise()
+
+            # append the metric for the step in the time partition
+            _metrics.append(_metric)
+
+        # set the first metric to the base
+        metric = _metrics[0]
+        # all the other metrics
+        metric.average(*_metrics[1:], weights=[dt] * len(_metrics))
```

Comment:
This is fine for now but I think we should implement a driver function for averaging on the Animate side to make this clearer. i.e., a function that effectively does this.

Reply:
See issue: https://github.com/mesh-adaptation/animate/issues/84.